etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,516 @@
+// We allow the lastUpdated timestamp to be empty on initial creation, but it is required once it has been set.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.lastUpdated) || (has(self.lastUpdated) && self.lastUpdated >= oldSelf.lastUpdated)",message="lastUpdated must be a newer timestamp than the current value"
+type PacemakerClusterStatus struct {
+	// lastUpdated is the timestamp when this status was last updated
+	// When present, it must be a valid timestamp in RFC3339 format and cannot be set to an earlier timestamp than the current value
+	// This is the most critical field in the status object because we can use it to warn if the status collection has gone stale.
+	// It is optional upon initial creation, but required once it has been set.
+	// +kubebuilder:validation:Format=RFC3339
+	// +optional
+	LastUpdated metav1.Time `json:"lastUpdated,omitempty"`
+
+	// rawXML contains the raw XML output from pcs status xml command.
+	// Kept for debugging purposes only; healthcheck should not need to parse this.
+	// When present, it must be between 1 and 262144 characters long (max 256KB). This limit protects the API from XML bombs and excessive memory consumption.
+	// When not present, the raw XML output is not available.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=262144
+	// +optional
+	RawXML string `json:"rawXML,omitempty"`
+
+	// collectionError contains any error encountered while collecting status
+	// When present, it must be between 1 and 2048 characters long (max 2KB). This limit ensures that the error message can be displayed in a human-readable format.
+	// When not present, no collection errors are available and the status collection is assumed to be successful.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	CollectionError string `json:"collectionError,omitempty"`
+
+	// summary provides high-level counts and flags for the cluster state
+	// TNF clusters don't use advanced pacemaker features like maintenance mode or pacemaker remote nodes, so these are omitted from the summary.
+	// When present, it must be a valid PacemakerSummary object.
+	// When not present, the summary is not available. This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	Summary *PacemakerSummary `json:"summary,omitempty"`
+
+	// nodes provides detailed information about each node in the cluster
+	// When present, it must be a list of 1 or 2 PacemakerNodeStatus objects. Two is expected in a healthy cluster.
+	// When not present, the nodes are not available. This likely indicates that there is an error parsing the raw XML output.
+	// If only one node is present, this indicates that the cluster is in the process of replacing a failed node.
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Nodes []PacemakerNodeStatus `json:"nodes,omitempty"`
```

Comment:
Will this functionality ever expand beyond just 2 nodes? Is there ever a chance there could be a 3rd node temporarily?